### ISSUE TITLE
Header self-refs should use anchor names chosen by the author

### ISSUE THREAD

**Issue by asbjornu** [ietf-tools/xml2rfc#1157]
### Describe the issue

As pointed out by @royfielding in https://github.com/ietf-tools/xml2rfc/issues/1156#issuecomment-2338999795:

> It is not appropriate for the tools to override the anchor names chosen by the authors. We deliberately chose those names and expect them to appear in all section and ToC references. It is still a bug even if the old references work, since we do not want two or three alias names for every section ref. We only want the ones that are authored in the XML.

As an example, the [Range Requests header in RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-range-requests) points to the anchor `#name-range-requests`, while the appropriate anchor to point to would be [`#range.requests`](https://www.rfc-editor.org/rfc/rfc9110.html#range.requests), as the presumably author-chosen identifier of the header's parent element:

```html
<div id="range.requests">
  <section id="section-14">
    <h2 id="name-range-requests">
      <a href="#section-14" class="section-number selfRef">14. </a>
      <a href="#name-range-requests" class="section-name selfRef">Range Requests</a>
    </h2>
    <p id="section-14-1">…</p>
    <p id="section-14-2">…</p>
    <div id="range.units">…</div>
    <div id="field.range">…</div>
    <div id="field.accept-ranges">…</div>
    <div id="field.content-range">…</div>
    <div id="partial.PUT">…</div>
    <div id="multipart.byteranges">…</div>
  </section>
</div>
```

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by reschke** [comment]
This is bad in particular as the document author has only limited control about what goes into the "slugifiedName" (as the algorithm for generating it is undefined).

**Comment by reschke** [comment]
It's a bug, as we want the author-provided anchors to be *used*. Those should be used by default (stability across revisions), and thus need to be what the readers see in the address bar (for bookmarking etc).

**Comment by rjsparks** [comment]
I don't think it's as trivial as you suggest - there are edges  - what do we do if there isn't an author provided tag for each section - should the ToC contain a _mix_ of author provided and auto-slugified hrefs targets? 

Note also that there's no way to discover the author provided tags unless they used in an internal link.

**Comment by reschke** [comment]
> I don't think it's as trivial as you suggest - there are edges - what do we do if there isn't an author provided tag for each section - should the ToC contain a _mix_ of author provided and auto-slugified hrefs targets?

Yes. That's what rfcxml.xslt has been doing for more than a decade. So has (IMHO) the predecessor of the current xml2rfc.

> Note also that there's no way to discover the author provided tags unless they used in an internal link.

That's what this ticket is about, no? The header "self-link" should link to the author-supplied anchor.

Check, for instance, https://greenbytes.de/tech/webdav/rfc9110.html - I guess that's where @royfielding was coming from. It happens that he was familiar with what my XSLT generates, because we used that for HTTP bis and tre.

**Comment by rjsparks** [comment]
To clarify - are you _only_ talking about ToC href values? (Where "you" could mean any of the thread participants.)
